packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -0,0 +1,667 @@
+// the settings plugin from the other ones
+const MenuItemDivider = () => (
+  <div className={styles['divider-first-item']}>
+    <div className={styles['divider-second-item']} />
+  </div>
+);
+MenuItemDivider.displayName = 'MenuItemDivider';
+
+// This component wraps `<RestrictedByPermissions>`
+// and the `<ToggleFeature>` components. However, it's necessary to have it as
+// the `<ToggleFeature>` wrapper should be rendered only if the `featureToggle`
+// prop is defined. This is because `<ToggleFeature>` will not render any
+// children if the flag is missing/not found.
+const hasMenuEveryMenuVisibilitySetToBeHidden = (
+  menuVisibilities?: TApplicationContext<{}>['menuVisibilities'],
+  namesOfMenuVisibilities?: string[]
+) =>
+  Array.isArray(namesOfMenuVisibilities) &&
+  namesOfMenuVisibilities.length > 0 &&
+  namesOfMenuVisibilities.every(
+    nameOfMenuVisibility =>
+      menuVisibilities && menuVisibilities[nameOfMenuVisibility] === true
+  );
+const isMenuItemDisabledForEnvironment = (
+  keyOfMenuItem: string,
+  disabledMenuItems?: string[]
+) => disabledMenuItems && disabledMenuItems.includes(keyOfMenuItem);
+
+type RestrictedMenuItemProps = {
+  featureToggle?: string;
+  namesOfMenuVisibilities?: string[];
+  menuVisibilities?: TApplicationContext<{}>['menuVisibilities'];
+  disabledMenuItems?: string[];
+  keyOfMenuItem: string;
+  permissions: string[];
+  actionRights?: TActionRight[];
+  dataFences?: TDataFence[];
+  children: React.ReactNode;
+};
+const restrictedMenuItemDefaultProps: Pick<
+  RestrictedMenuItemProps,
+  'permissions'
+> = {
+  permissions: [],
+};
+const RestrictedMenuItem = (props: RestrictedMenuItemProps) => {
+  // NOTE: Custom application are activated/deactivated while their
+  // visibility is not controlled via a visibiility overwrite.
+  if (
+    hasMenuEveryMenuVisibilitySetToBeHidden(
+      props.menuVisibilities,
+      props.namesOfMenuVisibilities
+    ) ||
+    isMenuItemDisabledForEnvironment(
+      props.keyOfMenuItem,
+      props.disabledMenuItems
+    )
+  )
+    return null;
+
+  const permissionsWrapper =
+    (Array.isArray(props.permissions) && props.permissions.length > 0) ||
+    (Array.isArray(props.dataFences) && props.dataFences.length > 0) ? (
+      <RestrictedByPermissions
+        permissions={props.permissions}
+        actionRights={props.actionRights}
+        dataFences={props.dataFences}
+        selectDataFenceData={demandedDataFence => {
+          switch (demandedDataFence.type) {
+            case 'store':
+              return demandedDataFence.actualDataFenceValues;
+            default:
+              return null;
+          }
+        }}
+        // Always check that some of the given permissions match.
+        shouldMatchSomePermissions={true}
+      >
+        {props.children}
+      </RestrictedByPermissions>
+    ) : (
+      <>{props.children}</>
+    );
+
+  if (props.featureToggle) {
```

Comment:
Nit: could acutally be a useFeatureToggle and ternary, right?

Reply:
That's tricky, cause the prop is optional, however the `useFeatureToggle` expects a flag value but because there is an early return in this component, declaring a conditional hook won't work. 🤷‍♂️